File: car_store_api/car_store_api/Models/Car.cs
using System.ComponentModel.DataAnnotations;

namespace car_store_api.Models
{
    public class Car
    {
        [Key]
        public int? Id { get; set; }

        [Required(ErrorMessage = "Name is required")]
        public string Name { get; set; }
        [Required(ErrorMessage = "Year is required")]
        public int Year { get; set; }

        [Required(ErrorMessage = "Color is required")]
        public string Color { get; set; }

        [Required(ErrorMessage = "Price is required")]
        [Range(1, 100000, ErrorMessage = "Price must be between $1 and $100000")]
        public double Price { get; set; }
        [Required(ErrorMessage = "SaleDate is required")]
        public DateTime SaleDate { get; set; }
    }
}

File: car_store_web/src/app/models/car.ts
class Car{
    id: number;
    name: string;
    year: number;
    color: string;
    price: number;
    saleDate: Date;
    constructor(id: number, name: string, year: number, color: string, price: number, saleDate: Date){
        this.id = id;
        this.name = name;
        this.year = year;
        this.color = color;
        this.price = price;
        this.saleDate = saleDate;
    }
}
export {Car};
